### PR TITLE
Clean up payments interface + router integration.

### PR DESCRIPTION
**Pull request by c00w**

### CODE REVIEW

File: payment.go [hunk 1]
```diff
@@ -10,15 +11,60 @@ import (
...
+	time   time.Time
+	amount int64
+}
+
+func payDebt(debts []debt, amount int64) []debt {
+	for _, d := range debts {
+		if d.amount < amount {
+			// not needed since removing
+			d.amount = 0
+			amount -= d.amount
```
Comment (eoswald) [inline payment.go:33]: What is the point of this?

Comment (c00w) [inline payment.go:33]: Utility function, since debt should be paid oldest to newest.

Comment (c00w) [inline payment.go:33]: Good catch, should happen in opposite order.

Comment (c00w) [inline payment.go:33]: Fixed.

**Pull request merged**